lib/augmentConfig.mjs

```diff
@@ -362,6 +389,52 @@ async function addPluginFunctions(config, { quietDeprecationWarnings }) {
 	return config;
 }
 
+/**
+ *
+ * @param {StylelintConfig} config
+ * @returns {Promise<StylelintConfig>}
+ */
+async function addProcessorFunctions(config) {
+	if (!config.processors) {
+		return config;
+	}
+
+	const processorPromises = config.processors.map(async (processorLookup) => {
+		let processorImport;
+
+		if (typeof processorLookup === 'string') {
+			processorImport = await dynamicImport(processorLookup);
+			processorImport = processorImport.default || processorImport;
+		} else {
+			processorImport = processorLookup;
+		}
+
+		const { name, ...functions } =
+			(typeof processorImport === 'function' ? await processorImport() : processorImport) ?? {};
+
+		if (!name) {
```

Comment:
[suggest] How about checking the `name` property?

```suggestion
		if (!isString(name) || !name) {
```

`isString` can be imported by:

```js
import { isString } from './utils/validateTypes.mjs';
```